tests/history/test_database.py

```
@@ -213,3 +213,107 @@ def test_row_factory(self, db_manager):
+            fh.truncate(size // 2)
+
+        db2 = DatabaseManager(temp_db_path)
+        try:
+            db2.initialize()
+        except DatabaseCorruptionError as exc:
+            msg = str(exc).lower()
+            # Must mention recovery/quarantine/move action so the caller
+            # can render a prompt. Exact wording locks in the contract.
+            assert any(word in msg for word in ("quarantine", "move aside", "rename", "back up")), (
+                f"error message lacks recovery guidance: {exc}"
+            )
```

Ruff (0.15.11)

[warning] Found assertion on exception `exc` in `except` block, use `pytest.raises()` instead

(PT017)